src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -349,40 +357,42 @@
-        public-key  (utils/get-public-key keypair)
-        private-key (utils/get-private-key keypair)
-        x500-name   (utils/cn (:ca-name ca-settings))
-        validity    (cert-validity-dates (:ca-ttl ca-settings))
-        serial      (next-serial-number! (:serial ca-settings))
-        ca-exts     (create-ca-extensions x500-name
-                                          serial
-                                          public-key)
-        cacert      (utils/sign-certificate
-                      x500-name
-                      private-key
-                      serial
-                      (:not-before validity)
-                      (:not-after validity)
-                      x500-name
-                      public-key
-                      ca-exts)
-        cacrl       (utils/generate-crl (.getIssuerX500Principal cacert)
-                                        private-key
-                                        public-key)]
-    (write-cert-to-inventory! cacert (:cert-inventory ca-settings))
-    (utils/key->pem! public-key (:capub ca-settings))
-    (utils/key->pem! private-key (:cakey ca-settings))
-    (utils/cert->pem! cacert (:cacert ca-settings))
-    (utils/crl->pem! cacrl (:cacrl ca-settings))))
+    (log/debug (str "Initializing SSL for the CA; settings:\n"
+                    (ks/pprint-to-string ca-settings)))
+    (create-parent-directories! (vals (settings->cadir-paths ca-settings)))
+    (-> ca-settings :csrdir fs/file ks/mkdirs!)
+    (-> ca-settings :signeddir fs/file ks/mkdirs!)
+    (initialize-serial-file! (:serial ca-settings))
+    (let [keypair     (utils/generate-key-pair keylength)
+          public-key  (utils/get-public-key keypair)
+          private-key (utils/get-private-key keypair)
+          x500-name   (utils/cn (:ca-name ca-settings))
+          validity    (cert-validity-dates (:ca-ttl ca-settings))
+          serial      (next-serial-number! (:serial ca-settings))
+          ca-exts     (create-ca-extensions x500-name
+                                            serial
+                                            public-key)
+          cacert      (utils/sign-certificate
+                        x500-name
+                        private-key
+                        serial
+                        (:not-before validity)
+                        (:not-after validity)
+                        x500-name
+                        public-key
+                        ca-exts)
+          cacrl       (utils/generate-crl (.getIssuerX500Principal cacert)
+                                          private-key
+                                          public-key)]
+      (write-cert-to-inventory! cacert (:cert-inventory ca-settings))
+      (utils/key->pem! public-key (:capub ca-settings))
+      (utils/key->pem! private-key (:cakey ca-settings))
+      (utils/cert->pem! cacert (:cacert ca-settings))
+      (utils/crl->pem! cacrl (:cacrl ca-settings)))))
```

Comment:
@cprice404 there's one extra indent because the function now has multiple parameter sets. But it's possible my whitespace settings are not the same as the original file one. Is there a convention?